OpenRA.Mods.Common/UpdateRules/Rules/20210321/AddControlGroups.cs

```diff
@@ -0,0 +1,29 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddControlGroups : UpdateRule
+	{
+		public override string Name => "Add new ControlGroups trait.";
+
+		public override string Description => "A new trait ControlGroups was added, splitting logic away from Selection.";
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			if (actorNode.ChildrenMatching("Selection").Any())
+				yield return "Add the ControlGroups trait to your World actor.";
```

Comment:
Would this work?

```c#
				if (actorNode.Key == "World" && !actorNode.ChildrenMatching("ControlGroups").Any())
						actorNode.AddNode(new MiniYamlNode("ControlGroups", ""));
```

Reply:
The key could also be `^BaseWorld` or something (idk why we have control groups in the EditorWorld) so probably not.

Reply:
but does editor world need it? If it doesn't then we don't mind not adding it to ^BaseWorld

Reply:
On second though this would add this `ControlGroups` to "World" overwrites as well, so this isn't the correct solution

Reply:
It also seems pretty weird to tie this rule to Selection trait